subprojects/logging/src/main/java/org/gradle/internal/logging/console/MultiLineBuildProgressArea.java

```diff
@@ -21,31 +21,23 @@
 import java.util.List;
 
 public class MultiLineBuildProgressArea implements BuildProgressArea {
-    private final List<DefaultRedrawableLabel> entries;
+    // 2 lines: 1 for BuildStatus and 1 for Cursor parking space
+    private final List<DefaultRedrawableLabel> entries = new ArrayList<DefaultRedrawableLabel>(2);
     private final DefaultRedrawableLabel progressBarLabel;
 
-    private final List<StyledLabel> buildProgressLabels;
+    private final List<StyledLabel> buildProgressLabels = new ArrayList<StyledLabel>();
+    private final DefaultRedrawableLabel parkingLabel;
     private final Cursor statusAreaPos = new Cursor();
     private boolean isVisible;
 
-    public MultiLineBuildProgressArea(int numLabels) {
-        this.buildProgressLabels = new ArrayList<StyledLabel>(numLabels);
-        // 2 extra lines: 1 for BuildStatus and 1 for Cursor parking space
-        this.entries = new ArrayList<DefaultRedrawableLabel>(numLabels + 2);
-
+    public MultiLineBuildProgressArea() {
         int row = 0;
 
         progressBarLabel = newLabel(row--);
         entries.add(progressBarLabel);
 
-        for (int i = 0; i < numLabels; ++i) {
-            DefaultRedrawableLabel label = newLabel(row--);
-            entries.add(label);
-            buildProgressLabels.add(label);
-        }
-
         // Parking space for the write cursor
-        entries.add(newLabel(row--));
+        entries.add(parkingLabel = newLabel(row--));
```

Comment:
It feels a bit terse to do assignment and list append on one line. Could you make this a bit more spelled out for old fogeys like me?